scripts/tools/DeliveryExecutionArchive.php

```diff
@@ -0,0 +1,270 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+namespace oat\taoDeliveryRdf\scripts\tools;
+
+use common_report_Report as Report;
+use oat\oatbox\action\ResolutionException;
+use oat\oatbox\extension\AbstractAction;
+use oat\taoDelivery\model\DeliverArchiveExistingException;
+use oat\taoDelivery\model\DeliveryArchiveNotExistingException;
+use oat\taoDeliveryRdf\model\DeliveryArchiveService;
+
+/**
+ * Run examples:
+ *
+ * - Show list of deliveries:
+ * ```
+ * sudo -u www-data php index.php 'oat\taoDeliveryRdf\scripts\tools\DeliveryExecutionArchive' list
+ * ```
+ */
+class DeliveryExecutionArchive extends AbstractAction
+{
+    /**
+     * @var array Available script modes
+     */
+    static public $options = ['list', 'archive', 'unarchive', 'delete'];
+
+    /**
+     * @var Report
+     */
+    protected $report;
+
+    /**
+     * @var array list of given params
+     */
+    protected $params;
+
+    /**
+     * @param $params
+     * @return Report
+     * @throws \common_exception_Error
+     */
+    public function __invoke($params)
+    {
+        $this->params = $params;
+
+        try {
+            $this->process();
+
+        } catch (\Exception $e) {
+            $this->helpAction($e->getMessage());
+        }
+
+        return $this->report;
+    }
+
+    /**
+     * Process action call
+     *
+     * @throws ResolutionException
+     * @throws \common_exception_Error
+     */
+    private function process()
```

Comment:
Could u check if the deliveryArchive service was registered before run this action?

Reply:
This method it's gonna throw \oat\oatbox\service\exception\InvalidServiceManagerException in case something it's not register, the exception it's catched in __invoke if something it's not register the exception message will be displayed.